src/patternfly/layouts/Flex/flex.scss

```diff
@@ -24,6 +25,21 @@ $pf-l-flex--variable-map: build-variable-map("pf-l-flex--spacer", $pf-l-flex--sp
   &:last-child {
     --pf-l-flex--spacer: 0;
   }
+
+  // set this var here so specificity is 20
```

Comment:
I really think we should leave these styles out of the flex layout. It goes against the principle that we have about the role of layouts vs components and how [layouts are used for positioning, and don't impose styles on their children](https://pf4.patternfly.org/guidelines#layouts), and I don't see a compelling reason for this exception in this PR. I would be in favor of keeping this in our back pocket and if there is a specific product requirement that makes this type of exception necessary, we can reconsider at that point.

Reply:
I feel strongly that we should make this exception now and consider `pf-c-divider` a special type of component, maybe the only of its type. In the preview, try putting the divider into a `pf-l-flex__item`. It does not behave as expected. We'd be better off to just use the existing separator. So, in order to use a component like `pf-c-divider` in place of all component specific divider/separators (in this case `pf-l-flex__separator`), all or most of them will require some change to the component/layout css for margin or placement updates, including the default, horizontal `divider`. 

In the case of `.pf-l-flex`, it would require a wrapper modifier, `.pf-l-flex__item.pf-m-divider` then placing `.pf-c-divider` inside that modified `flex item`. That to me seems redundant/inefficient, whereas we can just say, `.pf-c-divider` is a wildcard component and is the exception to the [the rule above](https://pf4.patternfly.org/guidelines#layouts). Since all layouts and components can use it, it gets special consideration. 

I also think our developers will appreciate this exception, as it's one less component specific element to use. If something needs to be divided, use `pf-c-divider` and it behaves as PF intended without worrying about how to configure the correct wrapping elements. Each component considers it and controls it specifically, if needed.

This approach is only an enhancement to the system. The only drawback is that it breaks the rules we wrote before we knew we needed to make exceptions to those rules.

Reply:
One more note: by using this method, `divider` responds to placement modifiers `.pf-m-space-items-{spacer}` applied to flex parents.  Which makes it responsive to the the built in spacing modifiers without having to explicitly adjust.

Reply:
> So, in order to use a component like pf-c-divider in place of all component specific divider/separators (in this case pf-l-flex__separator)

The flex layout doesn't have a separator, and at the point that we add one, it becomes a component at least by our current definition of a layout vs a component. 

> In the case of .pf-l-flex, it would require a wrapper modifier, .pf-l-flex__item.pf-m-divider then placing .pf-c-divider inside that modified flex item.

A known usage of the vertical variation is that the height of the parent needs to be managed in some way. You can do that now by using what's already built into the flex layout with `pf-m-align-self-stretch` if you wanted the divider to be a child of a flex item, or use the dividers `pf-m-stretch` if the divider is itself a flex child. https://codepen.io/mcoker/pen/abOBWgX

I get that it's an enhancement and could be beneficial, but it's inconsistent with the way layouts are built and we already have in place what we need to allow a divider to be used in a flex layout based off of the documentation in the divider component. I think using the modifier that already exists on the flex layout is consistent with how you would use the vertical divider in any other context. That's why I don't see there being a compelling reason to make an exception to the guidelines for this use case. 

At a minimum though I think this is beyond the scope of this PR of just adding the variation. It sounds like this might be a feature request to add dividers to the layouts, or the flex based ones at least. Maybe if this is a constant pain point for users we can consider making an exception and revisiting?